kernel/aster-nix/src/syscall/epoll.rs

```diff
@@ -109,9 +109,18 @@ fn do_epoll_wait(epfd: FileDesc, max_events: i32, timeout: i32) -> Result<Vec<Ep
         .get_file(epfd)?
         .downcast_ref::<EpollFile>()
         .ok_or(Error::with_message(Errno::EINVAL, "not epoll file"))?;
-    let epoll_events = epoll_file.wait(max_events, timeout.as_ref())?;
-
-    Ok(epoll_events)
+    let result = epoll_file.wait(max_events, timeout.as_ref());
+
+    // As the manual mentioned, the return value sould be zero if no file descriptor becomes ready
+    // during the requested `timeout` milliseconds. So we ignore Err(ETIME) and return empty vector.
+    //
+    // Manual: <https://www.man7.org/linux/man-pages/man2/epoll_wait.2.html>
+    if let Err(err) = result {
+        if err.error() == Errno::ETIME {
```

Comment:
Use [`Result::is_err_and`](https://doc.rust-lang.org/std/result/enum.Result.html#method.is_err_and).

Reply:
I have tried `Result::is_err_and` and it will consume the result. We have to return the result after ignoring the `Err(ETIME)`.

Reply:
I think the right pattern here is `result.as_ref().is_err_and(|err| ...)`?

Reply:
OK, thanks. Excuse my poor knowledge of the APIs.